osu.Game/Screens/Select/LocalScoreDeleteDialog.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Game.Overlays.Dialog;
+using osu.Game.Scoring;
+using System;
+using System.Threading.Tasks;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Logging;
+
+namespace osu.Game.Screens.Select
+{
+    public class LocalScoreDeleteDialog : PopupDialog
+    {
+        private ScoreManager scoreManager;
+
+        public LocalScoreDeleteDialog(ScoreInfo score)
+        {
+            try
+            {
+                string accuracy = string.Format(score?.Accuracy % 1 == 0 ? @"{0:P0}" : @"{0:P2}", score?.Accuracy);
+
+                BodyText = $@"{score} {Environment.NewLine} Rank: {score.Rank} - Max Combo: {score.MaxCombo} - {accuracy}";
+                Icon = FontAwesome.Solid.Eraser;
+                HeaderText = @"Clearing this local score. Are you sure?";
+                Buttons = new PopupDialogButton[]
+                {
+                    new PopupDialogOkButton
+                    {
+                        Text = @"Yes. Please.",
+                        Action = () => scoreManager.Delete(score)
+                    },
+                    new PopupDialogCancelButton
+                    {
+                        Text = @"No, I'm still attached.",
+                    },
+                };
+            }
+            catch (Exception e)
```

Comment:
This implementation is unnecessary. It's a situation which should never happen and is indicative of a developer error. This should be rewritten in the form:

```csharp
public LocalScoreDeleteDialog(ScoreInfo score)
{
    Debug.Assert(score != null);

    string accuracy = ...;
    ...
}
```

Reply:
Sounds good to me.

I wasn't sure whether doing an assertion or exception is better. 

I have updated with an assertion.